src/main/java/mate/academy/service/impl/MovieSessionServiceImpl.java

```diff
@@ -11,20 +11,33 @@
 @Service
 public class MovieSessionServiceImpl implements MovieSessionService {
     @Inject
-    private MovieSessionDao sessionDao;
+    private MovieSessionDao movieSessionDao;
 
     @Override
-    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
-        return sessionDao.findAvailableSessions(movieId, date);
+    public MovieSession add(MovieSession movieSession) {
+        return movieSessionDao.add(movieSession);
     }
 
     @Override
     public MovieSession get(Long id) {
-        return sessionDao.get(id).get();
+        return movieSessionDao.get(id).orElseThrow(
+                () -> new RuntimeException("movieSession not found"));
+    }
+
+    @Override
+    public List<MovieSession> findAvailableSessions(Long movieId, LocalDate date) {
+        return movieSessionDao.findAvailableSessions(movieId, date);
+    }
+
+    @Override
+    public Boolean update(MovieSession movieSession) {
+        MovieSession currentMovieSession = get(movieSession.getId());
+        movieSessionDao.update(movieSession);
+        return currentMovieSession.equals(get(movieSession.getId()));
     }
 
     @Override
-    public MovieSession add(MovieSession session) {
-        return sessionDao.add(session);
+    public Boolean delete(Long id) {
```

Comment:
Consider changing the return type of the `delete` method to `void`. Returning a boolean might not be necessary if the method is expected to always succeed or throw an exception otherwise.